src/staticanalysis/bot/static_analysis_bot/coverity/coverity.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import subprocess
+
+from cli_common.log import get_logger
+from static_analysis_bot import COVERITY
+from static_analysis_bot import AnalysisException
+from static_analysis_bot import DefaultAnalyzer
+from static_analysis_bot import Issue
+from static_analysis_bot import stats
+from static_analysis_bot.config import settings
+from static_analysis_bot.revisions import Revision
+
+logger = get_logger(__name__)
+
+ISSUE_MARKDOWN = '''
+## coverity error
+
+- **Message**: {message}
+- **Location**: {location}
+- **In patch**: {in_patch}
+- **Coverity check**: {check}
+- **Publishable **: {publishable}
+- **Is new**: {is_new}
+
+```
+{body}
+```
+'''
+
+
+class Coverity(DefaultAnalyzer):
+    '''
+    Coverity runner
+    '''
+    def __init__(self, validate_checks=True):
+        # Ensure that we have all what we need into settings
+        assert settings.cov_package_name, 'Missing secret COVERITY_CONFIG:package_name'
+        assert settings.cov_package_ver, 'Missing secret COVERITY_CONFIG:package_ver'
+
+        self.cov_state_path = os.path.join(os.environ['MOZBUILD_STATE_PATH'],
+                                           'coverity')
+        self.cov_path = os.path.join(self.cov_state_path,
+                                     settings.cov_package_name)
+        self.cov_run_desktop = os.path.join(self.cov_path, 'bin', 'cov-run-desktop')
+        self.cov_translate = os.path.join(self.cov_path, 'bin', 'cov-translate')
+        self.cov_configure = os.path.join(self.cov_path, 'bin', 'cov-configure')
+        self.cov_work_path = os.path.join(self.cov_state_path, 'data-coverity')
+        self.cov_idir_path = os.path.join(self.cov_work_path, settings.cov_package_ver, 'idir')
```

Comment:
Several remarks here why I didn't check this:

1. ```cov_translate``` and ```cov-configure``` are from taken from the tar that is provided by the server hence we trust that the 3rd party gives us sane data if the parent directory exists.
2. ```cov_work_path``` and ```cov_idir_path``` are not yet generated since they are temporary folders generated during the first step of the analysis when we run [desktop configuration](https://github.com/mozilla/release-services/pull/1849/commits/f0374159f6e40b309b9e1d245cd63f56c7a9919c#diff-6ef7211ccb05d1580484f66b237277b6R104), hence according to the documentation if that succeeds we have a guarantee that those folders do exist.  

Do you still want to have this checked with the new info?

Reply:
It's really up to you Andi. I generally do not trust external providers and double-check.